crates/amaru-ledger/src/state.rs

```diff
@@ -602,6 +602,7 @@ impl<S: Store, HS: HistoricalStores> State<S, HS> {
         &mut self,
         point: &Point,
         raw_block: &RawBlock,
+        arena_pool: &ArenaPool,
```

Comment:
I am not a *huge* fan of the drilling of the arena_pool through so many methods/functions. I just couldn't quite find the right place to have the arena_pool live. It is not state, it isn't quite the validation context either.

Would love some thoughts about the best place this arena pool should live.